test/Test/Prim/Int.purs

```diff
@@ -3,17 +3,32 @@ module Test.Prim.Int where
 import Prelude
 
 import Effect (Effect)
+import Test.QuickCheck (class Arbitrary)
+import Test.QuickCheck.Gen (chooseInt)
 import Test.QuickCheck.Laws (checkLaws)
 import Test.QuickCheck.Laws.Data as Data
 import Type.Proxy (Proxy(..))
 
+-- | Necessary for the EuclideanRing test
+-- | so as to prevent integer overflow when multiplying large integer values
+newtype EuclideanRingInt = EuclideanRingInt Int
+derive newtype instance eqEuclideanRingInt :: Eq EuclideanRingInt
+derive newtype instance semiringEuclideanRingInt :: Semiring EuclideanRingInt
+derive newtype instance ringEuclideanRingInt :: Ring EuclideanRingInt
+derive newtype instance commutativeRingEuclideanRingInt :: CommutativeRing EuclideanRingInt
+derive newtype instance euclideanRingEuclideanRingInt :: EuclideanRing EuclideanRingInt
+instance arbitraryEuclideanRingInt :: Arbitrary EuclideanRingInt where
+  arbitrary = do
+    EuclideanRingInt <$> chooseInt (-10_000) 10_000
```

Comment:
```suggestion
instance arbitraryEuclideanRingInt :: Arbitrary EuclideanRingInt where
  arbitrary = EuclideanRingInt <$> chooseInt (-10_000) 10_000
```

Is there a reason for `-10_000` to `10_000` specifically?

Reply:
Number.MAX_SAFE_INTEGER is `9007199254740991`. I figured 10k was a large enough range to test without being too large to produce the overflow.